packages/go_router/test/route_data_test.dart

```diff
@@ -181,6 +249,46 @@ void main() {
     );
   });
 
+  group('StatefulShellRouteData', () {
+    testWidgets(
+      'It should build the page from the overridden build method',
+      (WidgetTester tester) async {
+        final GoRouter goRouter = GoRouter(
+          initialLocation: '/child',
+          routes: <RouteBase>[
+            _statefulShellRouteDataBuilder,
+          ],
+        );
+        await tester.pumpWidget(MaterialApp.router(
+          routeInformationProvider: goRouter.routeInformationProvider,
```

Comment:
Use `MaterialApp.router(routerConfig: goRouter)` instead. Can you also convert any other place that you found doesn't use router config?